File: backend-rs/src/models/message.rs
use chrono::{DateTime, Utc};
use serde::Serialize;

#[derive(sqlx::FromRow, Serialize, Clone, Debug)]
pub struct ChatMessage {
    pub id: u64,
    pub channel_id: String,
    pub sender_id: String,
    pub sender_username: String,
    pub sender_display_name: String,
    pub sender_avatar_url: Option<String>,
    pub content: Option<String>,
    pub message_type: String,
    pub edited_at: Option<DateTime<Utc>>,
    pub reply_to_id: Option<String>,
    pub is_deleted: u8,
    pub created_at: DateTime<Utc>,
}

File: frontend/src/types/message.ts
export default interface Message {
  id: number;
  channel_id: string;
  sender_id: string;
  sender_username: string;
  sender_display_name: string;
  sender_avatar_url: string | null;
  content: string | null;
  message_type: string;
  edited_at: number;
  reply_to_id: string | null;
  is_deleted: number;
  created_at: number;
  attachments: Attachment[];
}

export interface Attachment {
  id: string;
  url: string;
  file_type: string;
  file_size: number;
  file_name: string;
  created_at: string;
}
